### ISSUE TITLE
files.download doesn't overwrite file

### ISSUE THREAD

**Issue by drewp** [Fizzadar/pyinfra#746]
I haven't tried to repro this. Hopefully the evidence from my shell history is useful.

Running this from host 'bang':
```
files.download(
        src=f'https://github.com/rancher/k3s/releases/download/{k3s_version}/{tail}',
        dest='/usr/local/bin/k3s',
        user='root',
        group='root',
        mode='755',
        cache_time=43000,
    )
```
said this:
```
    [dash] curl: (23) Failure writing output to destination
    [dash] Error
    [bang] Success
    [slash] Success
```
until I ran `sudo rm /usr/local/bin/skaffold` on dash. Then:
```
    [bang] No changes
    [slash] No changes
    [dash] Success
```
and now dash has this root-owned file:
```
-rwxr-xr-x 1 root root 60297216 Jan 31 23:20 /usr/local/bin/skaffold
```

I think dash should have worked the first time, like slash and bang did.

## Meta
+ Include output of `pyinfra --support`.

    System: Linux
      Platform: Linux-5.13.0-25-generic-x86_64-with-glibc2.34
      Release: 5.13.0-25-generic
      Machine: x86_64
    pyinfra: v1.6.1
    Executable: env/bin/pyinfra
    Python: 3.9.9 (CPython, GCC 10.3.0)

**Comment by drewp** [comment]
Still hitting this. I found this in strace on the machine that fails:

```
openat(AT_FDCWD, "/usr/local/bin/skaffold", O_WRONLY|O_CREAT|O_TRUNC, 0666) = -1 ETXTBSY (Text file busy)
```

Indeed, I have some running skaffold processes on dash. 

One fix is that pyinfra could download to another file and then `mv` it on top of the desired path.

**Comment by Fizzadar** [comment]
Thanks for digging into this @drewp - think the download then `mv` method is the best solution, aligns with how `files.put` works as well 👍.

**Comment by Fizzadar** [comment]
I just released `v1.6.2` which contains this change (https://github.com/Fizzadar/pyinfra/commit/5944d8e54252dca29d4dbf73592795d9cc13d871).
